src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs

```diff
@@ -100,64 +101,64 @@ public SolutionPreviewResult GetSolutionPreviews(Solution oldSolution, Solution
+                            await CreateChangedDocumentPreviewViewAsync(oldSolution.GetDocument(documentId), newSolution.GetDocument(documentId), zoomLevel, c).ConfigureAwaitRunInline()));
                     }
 
                     foreach (var documentId in projectChanges.GetAddedDocuments())
                     {
                         cancellationToken.ThrowIfCancellationRequested();
-                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, c =>
-                            CreateAddedDocumentPreviewViewAsync(newSolution.GetDocument(documentId), zoomLevel, c)));
+                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, async c =>
+                            await CreateAddedDocumentPreviewViewAsync(newSolution.GetDocument(documentId), zoomLevel, c).ConfigureAwaitRunInline()));
                     }
 
                     foreach (var documentId in projectChanges.GetRemovedDocuments())
                     {
                         cancellationToken.ThrowIfCancellationRequested();
-                        previewItems.Add(new SolutionPreviewItem(oldProject.Id, documentId, c =>
-                            CreateRemovedDocumentPreviewViewAsync(oldSolution.GetDocument(documentId), zoomLevel, c)));
+                        previewItems.Add(new SolutionPreviewItem(oldProject.Id, documentId, async c =>
+                            await CreateRemovedDocumentPreviewViewAsync(oldSolution.GetDocument(documentId), zoomLevel, c).ConfigureAwaitRunInline()));
                     }
 
                     foreach (var documentId in projectChanges.GetChangedAdditionalDocuments())
                     {
                         cancellationToken.ThrowIfCancellationRequested();
-                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, c =>
-                            CreateChangedAdditionalDocumentPreviewViewAsync(oldSolution.GetAdditionalDocument(documentId), newSolution.GetAdditionalDocument(documentId), zoomLevel, c)));
+                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, async c =>
+                            await CreateChangedAdditionalDocumentPreviewViewAsync(oldSolution.GetAdditionalDocument(documentId), newSolution.GetAdditionalDocument(documentId), zoomLevel, c).ConfigureAwaitRunInline()));
                     }
 
                     foreach (var documentId in projectChanges.GetAddedAdditionalDocuments())
                     {
                         cancellationToken.ThrowIfCancellationRequested();
-                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, c =>
-                            CreateAddedAdditionalDocumentPreviewViewAsync(newSolution.GetAdditionalDocument(documentId), zoomLevel, c)));
+                        previewItems.Add(new SolutionPreviewItem(documentId.ProjectId, documentId, async c =>
```

Comment:
Can we just change the constructor once rather than everybody else doing the workaround?

Reply:
No, the type getting constructed is in a different project and doesn't have access to `DifferenceViewerPreview`.

Reply:
Oh that's...funky. My first thought was to just move that to the WPF layer since it's clearly accepting WPF objects which makes me suspicious, but that pulls a lot more along for the ride. I don't like this but at this point it's not your fault you're only trying to make one part better. :smile: